src/span.h

```diff
@@ -27,6 +27,8 @@ class Span
     constexpr C* data() const noexcept { return m_data; }
     constexpr C* begin() const noexcept { return m_data; }
     constexpr C* end() const noexcept { return m_data + m_size; }
+    constexpr C& front() const noexcept { return m_data[0]; }
```

Comment:
No strong opinion but maybe assert in these?

Reply:
FWIW the stl version currently in gcc trunk doesn't assert anything(as usual in stl) https://godbolt.org/z/QNytT8

Reply:
for size checks? It's UB in stl so I figure just keep that behavior eh?

Reply:
I'd rather not add asserts here. `std::span` doesn't have them, so if we're expecting to someday replace `Span` with that, it should be subject to the same interface. And that interface requires the caller to make sure values are not out of range.

Spans may also be used in pretty tight loops some day, in which case asserts may be actually undesirable.

Reply:
Makes sense, you can resolve this.